src/test/start_server_test.ts

```diff
@@ -97,11 +98,20 @@ suite('startServer', () => {
       consoleWarn = console.warn;
     });
 
-    teardown(() => {
+    teardown((done) => {
       console.error = consoleError;
       console.warn = consoleWarn;
-      proxyServer && proxyServer.close();
-      app.close();
+      let waitingCount = 1;
+      function _done() {
+        if (--waitingCount === 0) {
+          done();
+        }
+      }
+      if (proxyServer) {
+        waitingCount++;
+        proxyServer.close(_done);
+      }
+      app.close(_done);
```

Comment:
I think we should strongly prefer async/await over callback counting for all the usual reasons.